cmd/antrea-agent/options.go

```diff
@@ -393,10 +391,15 @@ func (o *Options) setK8sNodeDefaultOptions() {
 	if o.config.HostProcPathPrefix == "" {
 		o.config.HostProcPathPrefix = defaultHostProcPathPrefix
 	}
-	if features.DefaultFeatureGate.Enabled(features.AntreaProxy) {
+	if o.config.AntreaProxy.Enable == nil {
+		o.config.AntreaProxy.Enable = pointer.Bool(true)
+	}
```

Comment:
IMO, maybe a good option would be to move the Feature gate validation code  from agent.go to this file:

```go
	if !features.DefaultFeatureGate.Enabled(features.AntreaProxy) {
		o.config.AntreaProxy.Enable = pointer.Bool(false)
		klog.InfoS("Feature gate `AntreaProxy` is deprecated, please use option `antreaProxy.enable` to disable AntreaProxy")
	} else if o.config.AntreaProxy.Enable == nil {
		o.config.AntreaProxy.Enable = pointer.Bool(true)
        }
```

This way, the rest of the code can ignore the Feature gate and just use `*o.config.AntreaProxy.Enable`. What do you think?

Reply:
Much better.